Python/Module.cpp

```diff
@@ -455,6 +455,15 @@ static void initPyHSPlasma_Constants(PyObject* module)
     PyModule_AddIntConstant(module, "KEY_UNMAPPED", KEY_UNMAPPED);
 }
 
+#if ((PY_MAJOR_VERSION == 3) && (PY_MINOR_VERSION < 9)) && defined(__GNUC__)
+#   define BEGIN_EXPORTED_DECLARATION _Pragma("GCC visibility push(default)")
+#   define END_EXPORTED_DECLARATION _Pragma("GCC visibility pop")
```

Comment:
Why not use an attribute (`__attribute__((visibility("default")))`) instead of needing to wrap the whole function body in begin/end macros?  The attribute macro can be placed after `PyMODINIT_FUNC` to ensure it takes priority over whatever `PyMODINIT_FUNC` defines...

Reply:
I tried that and it said the attribute was in the wrong spot and would be ignored.

`PyMODINIT_FUNC` is `extern "C" PyObject*`, and I think it wants the attribute before the `PyObject*` return type

Reply:
Ah, yeah, that does make it more annoying...  However, you could also forward-declare it with the attribute at the end:

```c++
PyMODINIT_FUNC PyInit_PyHSPlasma() __attribute__((visibility("default")));

PyMODINIT_FUNC PyInit_PyHSPlasma()
{
    ...
}
```

Reply:
Ahh okay, that's much cleaner.

Reply:
I'm not sure how "safe" it is to reuse `HSPLASMA_EXPORT` outside of the libHSPlasma core module itself, but it seems to work.

Reply:
I'm more worried about how "safe" it is to use in a location not documented as supported by CMake, since that macro now comes from CMake's GenerateExportHeader module.

Reply:
This seems like it would be safe to have unconditionally?

```cpp
PyMODINIT_FUNC PyInit_PyHSPlasma() [[gnu::visibility("default")]];
```

Python 3.7 already (as far as I can tell) annotates it with `__declspec(dllexport)` for MSVC

Reply:
Apparently `__attribute__((visibility("default")))` is the only thing that works... `[[gnu::visibility("default")]]` is ignored on non-class types 🙄